core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -75,6 +82,10 @@ void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQuerie
         } else if (sourceQuery instanceof ToParentBlockJoinQuery) {
             ToParentBlockJoinQuery blockJoinQuery = (ToParentBlockJoinQuery) sourceQuery;
             flatten(blockJoinQuery.getChildQuery(), reader, flatQueries, boost);
+        } else if (sourceQuery instanceof BoostingQuery) {
```

Comment:
If `sourceQuery` is `BoostingQuery`, need to flatten `BoostingQuery` 's `context` and `match` together.